trie/committer.go

```diff
@@ -108,7 +115,25 @@ func (c *committer) commitChildren(path []byte, n *fullNode) [17]node {
 		// Commit the child recursively and store the "hashed" value.
 		// Note the returned node can be some embedded nodes, so it's
 		// possible the type is not hashNode.
-		children[i] = c.commit(append(path, byte(i)), child)
+		if !parallel {
+			children[i] = c.commit(append(path, byte(i)), child, false)
+		} else {
+			wg.Add(1)
+			go func(index int) {
+				p := append(path, byte(i))
```

Comment:
shouldn't it be index?

`p := append(path, byte(index))`

Reply:
Well spotted! Yes, it should, because the way it's written not doesn't make sense. However, due to https://go.dev/blog/loopvar-preview - it means we can drop the `index` parameter and rely solely on `i`.

Reply:
> it means we can drop the `index` parameter and rely solely on `i`.

But, actually, I prefer to use `index` as an incoming parameter, I think it makes the code clearer